src/main/java/view/ListView.java

```diff
@@ -254,31 +252,36 @@ void removeSelectedMovie() {
      * Moves selected movie to a different list.
      */
     private void moveSelectedMovie() {
-        final int selectedRow = mediaListTable.getSelectedRow();
-        if (selectedRow == -1) {
-            System.err.println("No movie selected. Cannot move.");
+        if (mediaListTable == null || listViewModel == null || mediaCollectionSelector == null
+                || moveController == null) {
+            System.err.println("Initialization error. Required components are missing.");
         }
         else {
-            final ListState state = listViewModel.getState();
-            final String currentCollectionName = state.getCurrentCollectionName();
-            final Object movieNameObject = mediaListTable.getValueAt(selectedRow, 0);
-            if (movieNameObject == null) {
-                System.err.println("Movie name is null. Cannot move.");
+            final int selectedRow = mediaListTable.getSelectedRow();
+            if (selectedRow == -1) {
+                System.err.println("No movie selected. Cannot move.");
             }
             else {
-                final String movieName = movieNameObject.toString();
-                final String targetCollectionName = (String) mediaCollectionSelector.getSelectedItem();
-                if (targetCollectionName == null || targetCollectionName.equals(currentCollectionName)) {
-                    System.err.println("Invalid collection choice. Cannot move.");
+                final ListState state = listViewModel.getState();
+                final String currentCollectionName = state.getCurrentCollectionName();
+                final Object movieNameObject = mediaListTable.getValueAt(selectedRow, 0);
+                if (movieNameObject == null) {
+                    System.err.println("Movie name is null. Cannot move.");
                 }
                 else {
-                    moveController.moveMovie(currentCollectionName, targetCollectionName, movieName);
-                    System.out.println("Moved " + movieName + " from "
-                            + currentCollectionName + " to " + targetCollectionName);
+                    final String movieName = movieNameObject.toString();
+                    final String targetCollectionName = (String) mediaCollectionSelector.getSelectedItem();
```

Comment:
could add a null and type-check validation to ensure the selected item can be cast to String